app/src/main/java/org/mozilla/fenix/settings/TrackingProtectionFragment.kt

```diff
@@ -6,47 +6,113 @@ package org.mozilla.fenix.settings
 
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.lifecycleScope
 import androidx.navigation.Navigation
 import androidx.preference.Preference
 import androidx.preference.PreferenceFragmentCompat
 import androidx.preference.SwitchPreference
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
 import org.mozilla.fenix.R
 import org.mozilla.fenix.ext.getPreferenceKey
+import org.mozilla.fenix.ext.nav
 import org.mozilla.fenix.ext.requireComponents
 import org.mozilla.fenix.utils.Settings
 
 class TrackingProtectionFragment : PreferenceFragmentCompat() {
+
+    private lateinit var radioStrict: RadioButtonInfoPreference
+    private lateinit var radioStandard: RadioButtonInfoPreference
+
     override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
         setPreferencesFromResource(R.xml.tracking_protection_preferences, rootKey)
     }
 
     override fun onResume() {
         super.onResume()
-        activity?.title = getString(R.string.preferences_tracking_protection)
+        activity?.title = getString(R.string.preference_enhanced_tracking_protection)
         (activity as AppCompatActivity).supportActionBar?.show()
 
         // Tracking Protection Switch
         val trackingProtectionKey =
             context!!.getPreferenceKey(R.string.pref_key_tracking_protection)
         val preferenceTP = findPreference<SwitchPreference>(trackingProtectionKey)
 
-        preferenceTP?.isChecked = Settings.getInstance(context!!).shouldUseTrackingProtection
+        bindStrict()
+        bindRecommended()
+        setupRadioGroups()
+
+        val globalSetting = Settings.getInstance(context!!).shouldUseTrackingProtectionGlobal
+        preferenceTP?.isChecked = globalSetting
+
         preferenceTP?.onPreferenceChangeListener =
             Preference.OnPreferenceChangeListener { _, newValue ->
                 Settings.getInstance(requireContext())
-                    .setTrackingProtection(newValue = newValue as Boolean)
+                    .setTrackingProtectionGlobal(newValue = newValue as Boolean)
                 with(requireComponents) {
                     val policy = core.createTrackingProtectionPolicy(newValue)
                     useCases.settingsUseCases.updateTrackingProtection.invoke(policy)
                     useCases.sessionUseCases.reload.invoke()
                 }
+                radioStrict.isEnabled = newValue
+                radioStandard.isEnabled = newValue
                 true
             }
 
         val exceptions =
             context!!.getPreferenceKey(R.string.pref_key_tracking_protection_exceptions)
         val preferenceExceptions = findPreference<Preference>(exceptions)
         preferenceExceptions?.onPreferenceClickListener = getClickListenerForExceptions()
+        lifecycleScope.launch {
+            // TODO On first layout these views aren't inited yet for some reason
```

Comment:
Does anyone have any better solution here? On first inflate these views aren't available so can't be set enabled/disabled. The next onResume works fine...